services/wfm/src/main/java/org/openkilda/wfm/topology/isllatency/bolts/IslStatsBolt.java

```diff
@@ -57,39 +70,43 @@ public IslStatsBolt(String metricPrefix) {
-        tags.put("src_port", String.valueOf(data.getSource().getPortNo()));
-        tags.put("dst_switch", data.getDestination().getSwitchId().toOtsdFormat());
-        tags.put("dst_port", String.valueOf(data.getDestination().getPortNo()));
+        tags.put("src_switch", srcSwitchId.toOtsdFormat());
+        tags.put("src_port", String.valueOf(srcPort));
+        tags.put("dst_switch", dstSwitchId.toOtsdFormat());
+        tags.put("dst_port", String.valueOf(dstPort));
 
-        return tsdbTuple(metricFormatter.format("isl.latency"), timestamp, data.getLatency(), tags);
+        return tsdbTuple(metricFormatter.format("isl.latency"), timestamp, latency, tags);
     }
 
     @Override
     protected void handleInput(Tuple input) throws Exception {
-        Message message = pullValue(input, KafkaRecordTranslator.FIELD_ID_PAYLOAD, Message.class);
+        InfoData data = pullValue(input, LATENCY_DATA_FIELD, InfoData.class);
+        long timestamp = getCommandContext().getCreateTime();
 
-        if (message instanceof InfoMessage) {
-            InfoData data = ((InfoMessage) message).getData();
-            if (data instanceof IslInfoData) {
-                handleIslInfoData(input, message, (IslInfoData) data);
-            } else {
-                // There are much info data messages in kilda.topo.disco.storm topic.
-                // All of them except IslInfoData are useless for stats bolt so they will be ignored.
-                // It will be fixed by round trip latency feature. https://github.com/telstra/open-kilda/issues/580
-                // This feature adds separate topic for latency data.
-                // TODO: add unhandledInput(input) when round trip latency feature will be merged
-            }
+        if (data instanceof IslRoundTripLatency) {
+            Endpoint destination = pullValue(input, CACHE_DATA_FIELD, Endpoint.class);
+            islStatsService.handleRoundTripLatencyMetric(input, timestamp, (IslRoundTripLatency) data, destination);
+        } else if (data instanceof IslOneWayLatency) {
+            islStatsService.handleOneWayLatencyMetric(input, timestamp, (IslOneWayLatency) data);
         } else {
             unhandledInput(input);
         }
     }
 
-    private void handleIslInfoData(Tuple input, Message message, IslInfoData data) throws JsonEncodeException {
-        List<Object> results = buildTsdbTuple(data, message.getTimestamp());
-        getOutput().emit(input, results);
+    @Override
+    public void emitLatency(Tuple input, SwitchId srcSwitch, int srcPort, SwitchId dstSwitch, int dstPort,
+                            long latency, long timestamp) {
+        List<Object> tsdbTuple;
+        try {
+            tsdbTuple = buildTsdbTuple(srcSwitch, srcPort, dstSwitch, dstPort, latency, timestamp);
+        } catch (JsonEncodeException e) {
+            log.error(String.format("Couldn't create OpenTSDB tuple: %s", e.getMessage()), e);
```

Comment:
that's a bit wierd tbh, why not to log message properly at line 68 and do not rewrap JsonProcessingException into JsonEncodeException

Reply:
fixed